code/modules/admin/topic.dm

```diff
@@ -2782,9 +2782,11 @@
 			if("unpower")
 				feedback_inc("admin_secrets_fun_used",1)
 				feedback_add_details("admin_secrets_fun_used","UP")
-				log_admin("[key_name(usr)] made all areas unpowered", 1)
-				message_admins("<span class='notice'>[key_name_admin(usr)] made all areas unpowered</span>", 1)
-				power_failure()
+				if(alert("What Would You Like to Do?", "Make All Areas Unpowered", "Depower all APCs", "Short out APCs") == "Depower all APCs")
+					depower_apcs()
+				else
+					power_failure()
+				log_and_message_admins("<span class='notice'>[key_name_admin(usr)] made all areas unpowered</span>", 1)
```

Comment:
Can you make this log say they "shorted APCs" if they chose to short, and "made all areas unpowered" if they chose to de-power APCs?

Reply:
No problem. I was not sure what Admins would find most helpful here.

Side Note: A lot of the code here has both `log_admin` and `message_admins`.
`log_and_message_admins` exist and it's really ... really tempting to change those lines.

Reply:
`if(alert("What Would You Like to Do?", "Make All Areas Unpowered", "Depower all APCs", "Short out APCs") == "Depower all APCs")
					depower_apcs()
					log_and_message_admins("<span class='notice'>[key_name_admin(usr)] made All Areas Unpowered via Admin Secrets.</span>", 1)
				else
					power_failure()
					log_and_message_admins("<span class='notice'>[key_name_admin(usr)] has Shorted APCs via Admin Secrets.</span>", 1)`

Reply:
Additionally changed how the procs are called.